### ISSUE TITLE
CUDA 12.6 build failure

### ISSUE THREAD

**Issue by h6197627** [icl-utk-edu/magma#7]
MAGMA 2.8.0 building with CUDA 12.6 fails. At the same time building with CUDA 12.5 succeded.

I attached build error log: [log.txt](https://github.com/user-attachments/files/16637557/log.txt)

**Comment by abdelfattah83** [comment]
This should be resolved as of e8064dc

**Comment by jeffhammond** [comment]
Did you test this against CUDA 12.6?  I applied this change and it does not fix things.

**Comment by jeffhammond** [comment]
for context, i fixed all the min/max problems i saw because of these macros and the change set was huge:

 1874 files changed, 13180 insertions(+), 13174 deletions(-)

**Comment by jeffhammond** [comment]
Here are my patch files.  Do not use them because they contain global search and replace, including in comments and non-source files, but this is the changeset required for me to build with CUDA 12.6.

I recommend that you use C convention and name your macros MIN and MAX, and then use those globally.  I used std::min/max, which isn't ideal since I had to then replace cases to std::fmin/fmax where appropriate.

[0001-fix-fetch-macro-with-MIN.patch](https://github.com/user-attachments/files/18078291/0001-fix-fetch-macro-with-MIN.patch)
[0002-change-all-the-min-max-stuff.patch](https://github.com/user-attachments/files/18078292/0002-change-all-the-min-max-stuff.patch)
